src/vs/platform/extensions/common/extensions.ts

```diff
@@ -202,6 +202,39 @@ export interface IExtensionContributions {
 	readonly notebookRenderer?: INotebookRendererContribution[];
 }
 
+
+export function getImplicitActivationEvents(extensionContributions: IExtensionContributions): string[] {
```

Comment:
Is it worth creating some contribution model to have each main thread contribution consumer expose a mapping from event types to a handler that knows how to take the contribution and convert it to an implicit event? That would ensure that future contribution consumers don't need to find this function in order to opt into implicit events (and of course here I'm making the assumption that these contributions should result in implicit events being generated, which could have unforeseen consequences e.g. extensions getting activated when they don't need to be).